ts/components/session/RegistrationTabs.tsx

```diff
@@ -0,0 +1,581 @@
+            onSelect={this.handleTabSelect}
+          />
+        </div>
+        {this.renderSections()}
+      </div>
+    );
+  }
+
+  private readonly handleTabSelect = (event: TabSelectEvent): void => {
+    this.setState({ selectedTab: event.type });
+  };
+
+  private onSeedChanged(val: string) {
+    this.setState({ mnemonicSeed: val });
+  }
+
+  private onDisplayNameChanged(val: string) {
+    const sanitizedName = this.sanitiseNameInput(val);
+    this.setState({ displayName: sanitizedName });
+  }
+
+  private onPasswordChanged(val: string) {
+    this.setState({ password: val });
+    this.onValidatePassword(); // FIXME add bubbles or something to help the user know what he did wrong
+  }
+
+  private onPasswordVerifyChanged(val: string) {
+    this.setState({ validatePassword: val });
+  }
+
+  private renderSections() {
+    const { selectedTab } = this.state;
+    if (selectedTab === TabType.Create) {
+      return this.renderSignUp();
+    }
+
+    return this.renderSignIn();
+  }
+
+  private renderSignUp() {
+    const { signUpMode } = this.state;
+    const { i18n } = this.props;
+    if (signUpMode === SignUpMode.Default) {
+      return (
+        <div className="session-registration__content">
+          {this.renderSignUpHeader()}
+          {this.renderSignUpButton()}
+        </div>
+      );
+    } else {
+      return (
+        <div className="session-registration__content">
+          {this.renderSignUpHeader()}
+          <div className="session-registration__unique-session-id">
+            {i18n('yourUniqueSessionID')}
+          </div>
+          {this.renderEnterSessionID(false, this.state.hexEncodedPubKey)}
+          {this.renderSignUpButton()}
+          {this.getRenderTermsConditionAgreement()}
+        </div>
+      );
+    }
+  }
+
+  private getRenderTermsConditionAgreement() {
+    const { selectedTab, signInMode, signUpMode } = this.state;
+    if (selectedTab === TabType.Create) {
+      return signUpMode !== SignUpMode.Default
+        ? this.renderTermsConditionAgreement()
+        : null;
+    } else {
+      return signInMode !== SignInMode.Default
+        ? this.renderTermsConditionAgreement()
+        : null;
+    }
+  }
+
+  private renderSignUpHeader() {
+    const allUsersAreRandomly = this.props.i18n('allUsersAreRandomly...');
+
+    return <div className="session-signup-header">{allUsersAreRandomly}</div>;
+  }
+
+  private renderSignUpButton() {
+    const { signUpMode } = this.state;
+    const { i18n } = this.props;
+
+    let buttonType: any;
+    let buttonText: string;
+    if (signUpMode !== SignUpMode.Default) {
+      buttonType = SessionButtonType.FullGreen;
+      buttonText = i18n('getStarted');
+    } else {
+      buttonType = SessionButtonType.Green;
+      buttonText = i18n('generateSessionID');
+    }
+
+    return (
+      <SessionButton
+        onClick={() => {
+          if (signUpMode === SignUpMode.Default) {
+            void this.onSignUpGenerateSessionIDClick();
```

Comment:
as discussed with maxim we probably don't want this syntax to avoid confusion.
instead we want:
```js
this.onSignUpGenerateSessionIDClick().ignoreResult();
```

Reply:
yes!